apps/form.go

```diff
@@ -1,11 +1,39 @@
 package apps
 
+// Form defines what inputs a Call accepts, and how they can be gathered from
+// the user, in Modal and Autocomplete modes.
+//
+// For a Modal, the form defines the modal entirely, and displays it when
+// returned in response to a submit Call. Modals are dynamic in the sense that
+// they may be refreshed entirely when values of certain fields change, and may
+// contain dynamic select fields.
+//
+// For Autocomplete, a form can be bound to a sub-command. The behavior of
+// autocomplete once the subcommand is selected is designed to mirror the
+// functionality of the Modal. Some gaps and differences still remain.
+//
+// Requests for forms are calls, have expanded contexts, making it easy to
```

Comment:
"have expanded contexts" - is that a requirement?

Reply:
tweaked the language